thoth/adviser/boots/solvers_configured.py

```diff
@@ -0,0 +1,94 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""A boot to notify about runtime environments not supported by solvers enabled in a deployment."""
+
+import os
+import logging
+from typing import Any
+from typing import Dict
+from typing import Generator
+from typing import TYPE_CHECKING
+
+import attr
+from thoth.common import get_justification_link as jl
+from thoth.common import OpenShift
+from thoth.common.exceptions import SolverNameParseError
+
+from ..boot import Boot
+
+if TYPE_CHECKING:
+    from ..pipeline_builder import PipelineBuilderContext
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@attr.s(slots=True)
+class SolversConfiguredBoot(Boot):
+    """A boot to notify about runtime environments not supported by solvers enabled in a deployment."""
+
+    _SOLVERS_CONFIGURED = os.getenv("THOTH_ADVISER_DEPLOYMENT_CONFIGURED_SOLVERS")
+    _JUSTIFICATION = jl("eol_env")
+
+    @classmethod
+    def should_include(cls, builder_context: "PipelineBuilderContext") -> Generator[Dict[str, Any], None, None]:
+        """Register self if users use runtime environments not matching solvers in the configmap."""
+        if not cls._SOLVERS_CONFIGURED or builder_context.is_included(cls):
+            yield from ()
+            return None
+
+        runtime_environment = (
+            builder_context.project.runtime_environment.operating_system.name,
+            builder_context.project.runtime_environment.operating_system.version,
+            builder_context.project.python_version,
+        )
+        for solver_name in cls._SOLVERS_CONFIGURED.splitlines():
+            solver_name = solver_name.strip()
+
+            if not solver_name:
+                continue
+
+            try:
+                solver_info = OpenShift.parse_python_solver_name(solver_name)
+            except SolverNameParseError:
+                _LOGGER.exception("Please report this error to Thoth service administrator")
+                yield from ()
+                return None
+
+            if runtime_environment == (
+                solver_info["os_name"],
+                solver_info["os_version"],
+                solver_info["python_version"],
+            ):
+                break
+        else:
+            yield {}
+            return None
+
+        yield from ()
+        return None
+
+    def run(self) -> None:
+        """Inform about runtime environment that is not backed up by a solver."""
+        self.context.stack_info.append(
+            {
+                "type": "WARNING",
```

Comment:
shouldn't be error like https://thoth-station.ninja/j/solved_sw_env? Because we cannot provide recommendation for a solver we do not maintain or have data about. wdyt @fridex ?

Reply:
This is slightly different case. The `solved_sw_env` is captured by [SolvedSoftwareEnvironmentBoot](https://github.com/thoth-station/adviser/blob/66f1c7535c48796c0fc4d227b3917d5d1702afd1/thoth/adviser/boots/solved_software_environment.py#L39) pipeline unit.

This PR captures a gap when we have still old data in the database for solvers that are no longer enabled in the configmap. In such a case, we notify users that they use environments that Thoth administrator decommissioned and will be soon removed from the database.

Reply:
Should be documented in https://thoth-station.ninja/j/eol_env if that's not clear, let us know - I might have missed something.

Reply:
I understand now, sorry I missed the justification, I thought it was related to the case where all data from a solver decommissioned are no more there, which basically means we cannot provide advise on that runtime environment. Thanks!!